pkg/mapper/dynamicfile/dynamicfile.go

```diff
@@ -184,12 +185,18 @@ func (ms *DynamicFileMapStore) saveMap(
 	ms.awsAccounts = make(map[string]interface{})
 
 	for _, user := range userMappings {
-		canonicalizedARN, _ := arn.Canonicalize(strings.ToLower(user.UserARN))
-		ms.users[canonicalizedARN] = user
+		key, _ := arn.Canonicalize(strings.ToLower(user.UserARN))
+		if ms.userIDStrict {
+			key = user.UserId
```

Comment:
what happens if the config has entries without userId set, since it is optional regardless of whether `userIDStrict` is true/false?

should we throw error or warning?

For example, let's say i have 2 entries
- userArn:a
  userId:a-id
  username: a
- userArn:b
  username:b

the second entry will be useless if `userIDStrict` is true. Should user be warned that they have useless entries?

Reply:
i think logging a warning in case a useless entry is detected would be nice. But IDK how often this gets called so if it's annoying then never mind.


the inverse is also an issue but a much smaller one. If they have `userIDStrict` false and they have an entry with userId in it, actually userId will not be respected. So i don't think in this case warning is needed

Reply:
that's a good point. I will add the check for UserId when userIDStrict is true

Reply:
yeah I think its worth logging.  Also we can skip the lookup if its not set?